ntoskrnl/ke/i386/mp.c

```diff
@@ -0,0 +1,127 @@
+/*
+ * PROJECT:     ReactOS Kernel
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Architecture specific source file to hold multiprocessor functions
+ * COPYRIGHT:   Copyright 2021 Justin Miller <justinmiller100@gmail.com>
+ */
+
+/* INCLUDES *****************************************************************/
+
+#include <ntoskrnl.h>
+#define NDEBUG
+#include <debug.h>
+
+/* GLOBALS *******************************************************************/
+
+#define PHYSICAL_ADDRESS  LARGE_INTEGER
+PHYSICAL_ADDRESS HighestPhysicalAddress;
+ULONG ProcessorCount = 0;
```

Comment:
What is the difference between this variable and `StartedProcessorCount` from HAL?

Reply:
Well ones keeping track of how many times the KeStartAllProcessors loop has been ran, the other is to actually eventually associate an NTProcessorNumber to ApicID. 
Though this could be cleaned probably.